File: clojure/src/zulip_rest_api/specs/user_base.clj
(ns zulip-rest-api.specs.user-base
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            )
  (:import (java.io File)))


(def user-base-data
  {
   (ds/opt :email) string?
   (ds/opt :is_bot) boolean?
   (ds/opt :avatar_url) string?
   (ds/opt :avatar_version) int?
   (ds/opt :full_name) string?
   (ds/opt :is_admin) boolean?
   (ds/opt :is_owner) boolean?
   (ds/opt :is_billing_admin) boolean?
   (ds/opt :role) int?
   (ds/opt :bot_type) int?
   (ds/opt :user_id) int?
   (ds/opt :bot_owner_id) int?
   (ds/opt :is_active) boolean?
   (ds/opt :is_guest) boolean?
   (ds/opt :timezone) string?
   (ds/opt :date_joined) string?
   (ds/opt :delivery_email) string?
   (ds/opt :profile_data) (s/map-of string? any?)
   })

(def user-base-spec
  (ds/spec
    {:name ::user-base
     :spec user-base-data}))

File: fsharp-giraffe-server/OpenAPI/src/model/UserBase.fs
namespace OpenAPI.Model

open System
open System.Collections.Generic
open OpenAPI.Model.int option
open OpenAPI.Model.string option
open System.Collections.Generic

module UserBase = 

  //#region UserBase

  //#region enums
  type RoleEnum = _100Enum of int  |  _200Enum of int  |  _300Enum of int  |  _400Enum of int  |  _600Enum of int  
  //#endregion

  type UserBase = {
    Email : string;
    IsBot : bool;
    AvatarUrl : string option;
    AvatarVersion : int;
    FullName : string;
    IsAdmin : bool;
    IsOwner : bool;
    IsBillingAdmin : bool;
    Role : RoleEnum;
    BotType : int option;
    UserId : int;
    BotOwnerId : int option;
    IsActive : bool;
    IsGuest : bool;
    Timezone : string;
    DateJoined : string;
    DeliveryEmail : string;
    ProfileData : IDictionary<string, obj>;
  }
  //#endregion
  